src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -0,0 +1,186 @@
+import _ from 'lodash/fp'
+import { Fragment, useRef, useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { Link } from 'src/components/common'
+import { locationTypes } from 'src/components/region-common'
+import { updateRecentlyViewedWorkspaces } from 'src/components/workspace-utils'
+import { Ajax } from 'src/libs/ajax'
+import { responseContainsRequesterPaysError } from 'src/libs/ajax/ajax-common'
+import { saToken } from 'src/libs/ajax/GoogleStorage'
+import { getConfig } from 'src/libs/config'
+import { withErrorIgnoring, withErrorReporting } from 'src/libs/error'
+import { clearNotification, notify } from 'src/libs/notifications'
+import { useCancellation, useOnMount, useStore } from 'src/libs/react-utils'
+import { getUser, workspaceStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import { differenceFromNowInSeconds } from 'src/libs/utils'
+import { isAzureWorkspace, isGoogleWorkspace, WorkspaceWrapper } from 'src/libs/workspace-utils'
+import { defaultLocation } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+interface StorageDetails {
+  googleBucketLocation: string // historically returns defaultLocation if cannot be retrieved or Azure
+  googleBucketType: string // historically returns locationTypes.default if cannot be retrieved or Azure
+  azureContainerRegion: string|undefined
```

Comment:
Yes, it means it can be a string or undefined. I'm not sure what you mean about the ? suffix.

Reply:
I think what @aherbst-broad is suggesting is to use optional-property/member syntax here instead of string|undefined type... so
   `azureContainerRegion?: string`

Reply:
Ah, I wasn't aware of that. Isn't it sort of different…. does that not mean that `azureContainerRegion` may not exist at all, vs. existing with a value of `undefined`?

Reply:
Just read about it and now I understand.

_This is because TypeScript treats every optional property as a union type of undefined and assigned type._

Reply:
Fixed in https://github.com/DataBiosphere/terra-ui/pull/3765/commits/9b497d21ee80611de804955357a43e9ceb05a239